compiler/passes/src/type_checker/check_expressions.rs

```diff
@@ -175,11 +169,21 @@ impl<'a> TypeChecker<'a> {
 
                     // Allow `group` * `scalar` multiplication.
                     match (t1.as_ref(), t2.as_ref()) {
-                        (Some(Type::Group), Some(other)) | (Some(other), Some(Type::Group)) => {
+                        (Some(Type::Group), Some(other)) => {
```

Comment:
```suggestion
                        (Some(Type::Group), Some(other)) 
                        | (Some(other), Some(Type::Group))=> {
```
Small nit but I believe this can be simplified since the match blocks are the same.

Reply:
Ah, they shouldn't be because I need a different span for error generation depending on if it's the left or right that the group should be on. But that code was wrong and always used the left side span. Fixing that now.

Reply:
ah gotcha